gfx.cpp

```diff
@@ -3202,9 +3214,11 @@ class GfxInternal
         SetObjectName(gfx_dst, dst.name); SetObjectName(gfx_src, src.name);
         if(gfx_dst.resource_ == gfx_src.resource_)
             return GFX_SET_ERROR(kGfxResult_InvalidOperation, "Cannot copy between two buffer objects that are pointing at the same resource; use an intermediate buffer");
-        if(dst.cpu_access == kGfxCpuAccess_None) transitionResource(gfx_dst, D3D12_RESOURCE_STATE_COPY_DEST);
-        if(src.cpu_access == kGfxCpuAccess_None) transitionResource(gfx_src, D3D12_RESOURCE_STATE_COPY_SOURCE);
-        submitPipelineBarriers();   // transition our resources if needed
+        bool transitions = false;
+        if(dst.cpu_access == kGfxCpuAccess_None) transitions = transitionResource(gfx_dst, D3D12_RESOURCE_STATE_COPY_DEST, true);
+        if(src.cpu_access == kGfxCpuAccess_None) transitions |= transitionResource(gfx_src, D3D12_RESOURCE_STATE_COPY_SOURCE, true);
+        if(transitions)
+            submitPipelineBarriers();   // transition our resources if needed
```

Comment:
`transitionResource` is now updated to return a bool signalling if a transition is actually required. If not then no pipeline submission is required